models/sushiswap/base/sushiswap_base_trades.sql

```diff
@@ -0,0 +1,109 @@
+{{ config(
+    tags=['dunesql'],
+    schema = 'sushiswap_base',
+    alias = alias('trades'),
+    partition_by = ['block_month'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index'],
+    post_hook='{{ expose_spells(\'["base"]\',
+                                "project",
+                                "uniswap_v3",
+                                \'["wuligy"]\') }}'
+    )
+}}
+
+{% set project_start_date = '2023-07-25' %}
+
+WITH dexs AS
+(
+    --Sushiswap
+    SELECT
+        t.evt_block_time AS block_time
+        ,t.recipient AS taker
+        ,CAST(NULL as VARBINARY) as maker
+        ,CASE WHEN amount0 < INT256 '0' THEN abs(amount0) ELSE abs(amount1) END AS token_bought_amount_raw -- when amount0 is negative it means trader_a is buying token0 from the pool
+        ,CASE WHEN amount0 < INT256 '0' THEN abs(amount1) ELSE abs(amount0) END AS token_sold_amount_raw
+        ,NULL AS amount_usd
+        ,CASE WHEN amount0 < INT256 '0' THEN f.token0 ELSE f.token1 END AS token_bought_address
+        ,CASE WHEN amount0 < INT256 '0' THEN f.token1 ELSE f.token0 END AS token_sold_address
+        ,t.contract_address as project_contract_address
+        ,t.evt_tx_hash AS tx_hash
+        ,t.evt_index
+    FROM
+    {{ source('sushiswap_v3_base', 'UniswapV3Pool_evt_Swap') }} t
+    INNER JOIN 
+    {{ source('sushiswap_base', 'UniswapV3Factory_evt_PoolCreated') }} f
+        ON f.pool = t.contract_address
+    {% if is_incremental() %}
+    WHERE t.evt_block_time >= date_trunc('day', now() - interval '7' day)
+    {% endif %}
+)
+SELECT
+    'base' AS blockchain
+    ,'sushiswap' AS project
+    ,'1' AS version
+    ,TRY_CAST(date_trunc('DAY', dexs.block_time) AS date) AS block_date
+    ,cast(date_trunc('month',dexs.block_time) as date) as block_month
+    ,dexs.block_time
+    ,erc20a.symbol AS token_bought_symbol
+    ,erc20b.symbol AS token_sold_symbol
+    ,case
+        when lower(erc20a.symbol) > lower(erc20b.symbol) then concat(erc20b.symbol, '-', erc20a.symbol)
+        else concat(erc20a.symbol, '-', erc20b.symbol)
+    end as token_pair
+    ,dexs.token_bought_amount_raw / power(10, erc20a.decimals) AS token_bought_amount
+    ,dexs.token_sold_amount_raw / power(10, erc20b.decimals) AS token_sold_amount
+    ,CAST(dexs.token_bought_amount_raw AS UINT256) AS token_bought_amount_raw
+    ,CAST(dexs.token_sold_amount_raw AS UINT256) AS token_sold_amount_raw
```

Comment:
can we double check out sushiswap blockchain spells (or even uniswap) to see what data types the amount fields use?

if you're casting lines 58/59, which is correct, i wonder if  you will need to cast 56/57 as well to `uint256` or maybe another data type

Reply:
the data explorer should help us see types